src/app/component/MyAccountMyOrders/MyAccountMyOrders.component.js

```diff
@@ -13,15 +13,251 @@ import { PureComponent } from 'react';
+        return __('Order history');
+    }
+
+    static getTabPathname() {
+        return ORDERS_PATH;
+    }
+
+    static isVisible() {
+        return true;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.shownOrderValues = [
+            'increment_id',
+            'status_label',
+            'created_at',
+            'grand_total'
+        ];
+
+        this.tableHeadingRow = {
+            [ID]: `${__('Order')} №`,
+            [CREATED_AT]: __('Date'),
+            [STATUS_LABEL]: __('Status Label'),
+            [GRAND_TOTAL]: __('Order - Total'),
+            [SHOW_MORE]: __('Action')
+        };
+
+        this.renderOrderTableRow = this.renderOrderTableRow.bind(this);
+    }
+
+    componentDidMount() {
+        const {
+            getOrderList, getOrderById, isMobileTab, orderId
+        } = this.props;
+
+        if (isMobileTab) return;
+
+        if (orderId === ORDERS_PATH) {
+            getOrderList();
+        } else {
+            getOrderById(orderId);
+        }
+    }
+
+    processOrderItem(key, value, is_b2b) {
+        if (!this.shownOrderValues.includes(key)) {
+            return null;
+        }
+
+        switch (key) {
+        case CREATED_AT:
+            const { getFormattedDate } = this.props;
+            const formattedDate = getFormattedDate(value);
+
+            return this.renderOrderTableColumn(key, formattedDate, null, is_b2b);
+        case GRAND_TOTAL:
+            const priceString = formatCurrency(value);
+
+            return this.renderOrderTableColumn(key, priceString, null, is_b2b);
+        default:
+            return this.renderOrderTableColumn(key, value, null, is_b2b);
+        }
+    }
+
+    renderHeading() {
+        return (
+            <h3
+              block="MyAccountMyOrders"
+              elem="Heading"
+            >
+                { __('Order history') }
+            </h3>
+        );
+    }
+
+    renderOrderTable() {
+        const { orderList } = this.props;
+
+        const items = orderList.length
+            ? orderList
+            : Array.from({ length: 7 }, (_, i) => ({ base_order_info: { id: i } }));
+
+        return (
+            <ul
+              block="MyAccountMyOrders"
+              elem="List"
+            >
+                { this.renderHeadingRow() }
+                { items.map(order => this.renderOrderTableRow(order)) }
+            </ul>
+        );
+    }
+
+    renderHeadingRow() {
+        const { isMobileTab } = this.props;
+
+        if (isMobileTab) return null;
+
+        return (
+            <li
+              block="MyAccountMyOrders"
+              elem="Row"
+            >
+                <ul
+                  block="MyAccountMyOrders"
+                  elem="RowItems"
+                  mods={ { type: 'heading' } }
+                >
+                    { Object.entries(this.tableHeadingRow)
+                        .map(([key, value]) => this.renderOrderTableColumn(key, value, 'heading')) }
```

Comment:
You are not allowed to use bind in render, you must create a separate component for this.